internal/edge_cluster/edge_node_subresource.go

```diff
@@ -0,0 +1,137 @@
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The administrator password for the edge node",
+			},
+			"audit_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The audit password for the edge node",
+			},
+			"root_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The root user password for the edge node",
+			},
+			"tep1_ip": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The IP address (CIDR) of the first tunnel endpoint",
+			},
+			"tep2_ip": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The IP address (CIDR) of the second tunnel endpoint",
+			},
+			"tep_gateway": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The gateway for the tunnel endpoints",
+			},
+			"tep_vlan": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "The VLAN ID for the tunnel endpoint",
+			},
+			"inter_rack_cluster": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether or not this is an inter-rack cluster. True for L2 non-uniform and L3, false for L2 uniform",
+			},
+			"management_ip": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The IP address (CIDR) for the management network",
+			},
+			"management_gateway": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The gateway for the management network",
+			},
+			"first_nsx_vds_uplink": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the first NSX-enabled VDS uplink",
+			},
+			"second_nsx_vds_uplink": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of the second NSX-enabled VDS uplink",
+			},
+			"uplink": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Specifications of Tier-0 uplinks for the edge node",
+				Elem:        UplinkNetworkSchema(),
+			},
+		},
+	}
+}
+
+func UplinkNetworkSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"interface_ip": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The IP address (CIDR) for the distributed switch uplink",
+			},
+			"vlan": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "The VLAN ID for the distributed switch uplink",
+			},
+			"bgp_peer": {
+				Type:        schema.TypeList,
+				Required:    false,
+				Optional:    true,
+				Description: "List of BGP Peer configurations",
+				Elem:        BgpPeerSchema(),
+			},
+		},
+	}
+}
+
+func BgpPeerSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"ip": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "IP address",
+			},
+			"password": {
```

Comment:
What are the constraints on this password?

Reply:
There are no constraints on BGP peer passwords. I'm going to set NoZeroValues on this one